sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -97,6 +97,109 @@ kernel_pid_t gnrc_ipv6_init(void)
     return gnrc_ipv6_pid;
 }
 
+static bool _dispatch_insert_ext_headers(gnrc_pktsnip_t *pkt)
+{
+#ifdef MODULE_GNRC_IPV6_EXT
+    DEBUG("ipv6: dispatch insertion of extension headers\n");
+    uint8_t ext_demux[] = { PROTNUM_IPV6_EXT_HOPOPT
+                          , PROTNUM_IPV6_EXT_RH
+                          , PROTNUM_IPV6_EXT_FRAG
+                          , PROTNUM_IPV6_EXT_ESP
+                          , PROTNUM_IPV6_EXT_AH
+                          , PROTNUM_IPV6_EXT_DST
+                          , PROTNUM_IPV6_EXT_MOB
+                          };
+    bool encapsulate = false;
+
+    for ( size_t i = 0; i < sizeof(ext_demux); ++i ) {
+        gnrc_netreg_entry_t *call = gnrc_netreg_lookup(GNRC_NETTYPE_IPV6, ext_demux[i]);
+
+        while (call) {
+            if (!encapsulate && (ext_demux[i] == PROTNUM_IPV6_EXT_HOPOPT
+                || ext_demux[i] == PROTNUM_IPV6_EXT_RH) ) {
+                encapsulate = true;
+            }
+            gnrc_pktsnip_t *data = pkt;
+            gnrc_netapi_get(call->target.pid, NETOPT_IPV6_EXT_HDR, ext_demux[i],
+                            data, sizeof(gnrc_pktsnip_t *));
+
+            if (data) {
+                /* insert the extension header snip */
+                ipv6_hdr_t* hdr = gnrc_ipv6_get_header(pkt);
+                if (hdr && (pkt->data == (void *)hdr)) {
+                    /* data has been placed just below the IPv6 Header,
+                     * i.e. in front of hdr->next.
+                     * We bend hdr->next to point to ext.
+                     */
+                    pkt->next = data;
```

Comment:
It depends how the pktsnip is created. 
When we create it with the pkt prepending to pkt->next I would expect that the former content is appended behind the newly created snip.
e.g. `gnrc_pktbuf_add(pkt->next, ....);`